crates/server/src/lib.rs

```diff
@@ -108,14 +111,42 @@ pub async fn start(
                 http::Method::DELETE,
                 http::Method::PUT,
                 http::Method::OPTIONS,
-            ]),
+            ])
+            .allow_private_network(true)
     );
+    // Check if the certificate exists and if they contain the current local IP address
+    certificates::check_for_certificate()?;
+
+    // Get the certificate directory and the paths to the certificate and private key
+    let certificate_dir = certificates::get_certificate_dir();
+    let cert_path = certificate_dir.join("cert.pem");
+    let key_path = certificate_dir.join("key.pem");
+
+    // Configure certificate and private key used by https
+    let rustls_config = match RustlsConfig::from_pem_file(cert_path, key_path).await {
+        Ok(config) => config,
+        Err(e) => {
+            eprintln!("Failed to load TLS configuration: {:?}", e);
+            return Err(e.into());
+        }
+    };
 
     let mut set = tokio::task::JoinSet::new();
 
     for listener in listeners {
+        let rustls_config = rustls_config.clone();
         let app = app.clone();
-        set.spawn(async { axum::serve(listener, app).await });
+        let addr = listener.local_addr().unwrap();
+        set.spawn(async move {
+            if let Err(e) = axum_server::bind_rustls(addr, rustls_config)
```

Comment:
looks like this will only permit TLS-able requests, we still need to conditionally allow HTTP right?

or are we forbidding that, favoring HTTPS always?

There are some options for this:

- https://crates.io/crates/axum-server-dual-protocol
- https://users.rust-lang.org/t/how-to-serve-http-and-https-on-the-same-port-with-hyper/103759/2